models/migrations/v184.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"xorm.io/xorm"
+)
+
+func dropWebhookColumns(x *xorm.Engine) error {
+	// Make sure the columns exist before dropping them
+	type Webhook struct {
+		Signature string `xorm:"TEXT"`
+		IsSSL     bool   `xorm:"is_ssl"`
+	}
+	if err := x.Sync2(new(Webhook)); err != nil {
+		return err
+	}
+
+	type HookTask struct {
+		Typ         string `xorm:"VARCHAR(16) index"`
+		URL         string `xorm:"TEXT"`
+		Signature   string `xorm:"TEXT"`
+		HTTPMethod  string `xorm:"http_method"`
+		ContentType int
+		IsSSL       bool
+	}
+	if err := x.Sync2(new(HookTask)); err != nil {
+		return err
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+	if err := dropTableColumns(sess, "webhook", "signature", "is_ssl"); err != nil {
+		return err
+	}
+	if err := dropTableColumns(sess, "hook_task", "typ", "url", "signature", "http_method", "content_type", "is_ssl"); err != nil {
```

Comment:
don't know why this columns should be dropped? webhook maybe changed after hook_task finished. So these columns will keep the original information when requesting.

Reply:
**Webhook**:
`Signature` and `IsSSL` are never used

**HookTask**:
`Typ` see https://github.com/go-gitea/gitea/pull/16176#discussion_r653250179
`URL` moved to `HookRequest`
`Signature` calculated on delivery. Is saved indirect in the `Header` map in `HookRequest`
`HTTPMethod` moved to `HookRequest`
`ContentType` of the Webhook is used on delivery. If the Webhook gets changed before sending the HookTask will use the current/correct settings. The original value is not available after a change but it's not visible on the UI anyway.
`IsSSL` is never used